accumulate/example.java

```diff
@@ -0,0 +1,17 @@
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+public class Accumulate {
+
+	public static <T> Collection<T> accumulate(Collection<T> collection, Function<T, T> function) {
```

Comment:
Yeah, in the Ruby standard library this behavior is called both `map` and `collect` (one is an alias of the other). I'm not sure why I chose `accumulate` as the name for this :/